renku/core/models/datasets.py

```diff
@@ -42,6 +45,14 @@
 )
 
 
+def str_or_doi(value):
```

Comment:
Isn't this gonna return string type in both cases? Maybe we should come up with a better name?

Reply:
well it casts to string or returns the doi (which yes, is also a string). Originally the converter was just `str`.

Reply:
Yes, hence the confusion. Maybe better name would be `origin_or_doi`?

Reply:
I could rename it though I don't know that those names would make it clearer - a converter of just `str` is not confusing?

Reply:
Well no, cause we are only talking about typing with that converter. Here we are mixing semantics of logic with typing, let's go with only 1. If we would have DOI type then `str_or_doi` would make sense and thats what I would expect with this name, but since this is not the case this makes it harder to reason about it when reading it.

Reply:
ok, renamed to `_extract_doi`